src/handlers/records-query.ts

```diff
@@ -216,20 +230,26 @@ export class RecordsQueryHandler implements MethodHandler {
   }
 
   /**
-   * Does the recordQuery filter explicitly for published records.
-   * Checks that there is either an explicit datePublished filter, or published is explicitly not set to false.
+   * Checks if the recordQuery filter supports returning published records.
    */
   private static filtersForPublishedRecords(recordsQuery: RecordsQuery): boolean {
     const { filter } = recordsQuery.message.descriptor;
-    return filter.datePublished !== undefined || filter.published !== false;
+    // When `published` and `datePublished` range are both undefined, published records can be returned.
+    if (filter.datePublished === undefined && filter.published === undefined) {
+      return true;
+    }
+    return filter.datePublished !== undefined || filter.published === true;
```

Comment:
Sanity: why is the original implementation not sufficient?

```suggestion
    return filter.datePublished !== undefined || filter.published !== false;
```

Reply:
It was, and since the comment is there it shows more clear intent so reverted back to original implementation.